packages/vite/src/executors/dev-server/dev-server.impl.ts

```diff
@@ -0,0 +1,233 @@
+import { ExecutorContext, joinPathFragments, logger } from '@nrwl/devkit';
+
+import 'dotenv/config';
+import {
+  InlineConfig,
+  UserConfig,
+  UserConfigExport,
+  ProxyOptions,
+  createServer,
+  mergeConfig,
+  ViteDevServer,
+  searchForWorkspaceRoot,
+  defineConfig,
+} from 'vite';
+import { join, relative } from 'path';
+import { existsSync } from 'fs';
+import react from '@vitejs/plugin-react';
```

Comment:
Make sure React or framework-specific plugins are installed to workspace, and used in `vite.config.ts`, so we don't need to provide much in terms of built-in configuration.